### PR TITLE
Make SessionManager able to unregister

### PR DESCRIPTION
**Pull request by trixon**
So the goal here is to add unregister methods and write to preferences during registration in order to use SessionManager for user custom presets.

### CODE REVIEW

**Comment by trixon** [comment]
When I was about to add the missing javadoc I noticed that the methods were missing too...

File: gemsfx/src/main/java/com/dlsc/gemsfx/util/SessionManager.java [hunk 1]
```diff
@@ -63,13 +68,15 @@ public void register(String path, DoubleProperty property) {
     public void register(String path, IntegerProperty property) {
         LOG.fine("registering integer property at path " + path);
         property.set(preferences.getInt(path, property.get()));
-        property.addListener((it, oldValue, newValue) -> {
+        initializePreferences(path, property);
+        var listener = (ChangeListener<Number>) (it, oldValue, newValue) -> {
```
Comment (dlemmermann) [inline gemsfx/src/main/java/com/dlsc/gemsfx/util/SessionManager.java:72]: We do not use the var keyword in GemsFX.

File: gemsfx/src/main/java/com/dlsc/gemsfx/util/SessionManager.java [hunk 2]
```diff
@@ -143,12 +156,62 @@ public void register(String path, BooleanProperty property) {
...
+        }
+        propertyToListeners.clear();
+    }
+
+    private void addListener(Property property, ChangeListener listener) {
+        property.addListener(listener);
+        propertyToListeners.computeIfAbsent(property, k -> new ArrayList<>()).add(listener);
+    }
+
+    private void initializePreferences(String path, Property property) {
```
Comment (dlemmermann) [inline gemsfx/src/main/java/com/dlsc/gemsfx/util/SessionManager.java:201]: Not sure I like the name. Maybe "putPropertyValueIntoPreferences"

**Review by dlemmermann — CHANGES_REQUESTED** [review]

File: gemsfx/src/main/java/com/dlsc/gemsfx/util/SessionManager.java (hunk 1 above)
Comment (trixon) [reply gemsfx/src/main/java/com/dlsc/gemsfx/util/SessionManager.java]: Fixed. May  I ask why?

File: gemsfx/src/main/java/com/dlsc/gemsfx/util/SessionManager.java (hunk 2 above)
Comment (trixon) [reply gemsfx/src/main/java/com/dlsc/gemsfx/util/SessionManager.java]: Fixed.

**Review by dlemmermann — APPROVED** [review]

File: gemsfx/src/main/java/com/dlsc/gemsfx/util/SessionManager.java (hunk 1 above)
Comment (dlemmermann) [reply gemsfx/src/main/java/com/dlsc/gemsfx/util/SessionManager.java]: Convention ..... as in "I do not use it". I prefer reading declarations left to right. Just what I am used to.

**Pull request merged**